include/SDL3/SDL_process.h

```diff
@@ -0,0 +1,180 @@
+ * \returns the newly created and running process, or NULL if the process couldn't be created.
+ *
+ * \sa SDL_CreateProcessWithProperties
+ * \sa SDL_GetProcessProperties
+ * \sa SDL_KillProcess
+ * \sa SDL_WaitProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_Process *SDLCALL SDL_CreateProcess(const char * const *args);
+
+/**
+ * Create a new process with the specified properties.
+ *
+ * These are the supported properties:
+ *
+ * - `SDL_PROP_PROCESS_CREATE_ARGS_POINTER`: an array of strings containing the program to run, any arguments, and a NULL pointer, e.g. const char *args[] = { "myprogram", "argument", NULL }. This is a required property.
+ * - `SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER`: an array of strings containing variable=value, and a NULL pointer, e.g. const char *env[] = { "PATH=/bin:/usr/bin", NULL }. If this property is set, it will be the entire environment for the process, otherwise the current environment is used.
+ * - `SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN`: true if an SDL_IOStream should be created as input to the process and available as `SDL_PROP_PROCESS_STDIN_POINTER` in the properties for the created process. By default the process has no input.
+ * - `SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN`: true if an SDL_IOStream should be created for output from the process and available as `SDL_PROP_PROCESS_STDOUT_POINTER` in the properties for the created process. By default the process shares the output of the current application.
+ * - `SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN`: true if an SDL_IOStream should be created for error output from the process and available as `SDL_PROP_PROCESS_STDERR_POINTER` in the properties for the created process. By default the process shares the error output of the current application.
+ * - `SDL_PROP_PROCESS_CREATE_REDIRECT_STDERR_BOOLEAN`: true if the error output of the process should be redirected into the standard output of the process. This property has no effect if `SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN` is true.
+ *
+ * On POSIX platforms, waitpid(-1, ...) should not be called, and SIGCHLD should not be ignored or handled because it would prevent SDL from properly tracking the lifetime of the underlying process. You should use SDL_WaitProcess() instead.
+ *
+ * \param props the properties to use.
+ * \returns the newly created and running process, or NULL if the process couldn't be created.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_GetProcessProperties
+ * \sa SDL_KillProcess
+ * \sa SDL_WaitProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_Process *SDLCALL SDL_CreateProcessWithProperties(SDL_PropertiesID props);
+
+#define SDL_PROP_PROCESS_CREATE_ARGS_POINTER            "SDL.process.create.args"
+#define SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER     "SDL.process.create.environment"
```

Comment:
Would a `SDL_PROP_PROCESS_CREATE_CWD_POINTER` property make sense?
Or does https://github.com/libsdl-org/SDL/issues/9110 still apply?

Platforms that don't know about a cwd concept, can ignore the property.
The "global state" motivation does not apply.

Reply:
Yes, it does, but this can be added after this PR lands. I'm also not sure how it would be implemented with posix_spawn().

Reply:
It's not possible atm.

Reply:
It's horrible, but setting the argv to (in Python-like pseudocode)

```
['/bin/sh', '-ec', 'cd "$1"; shift; exec "$@"', 'sh', desired_cwd] + original_argv
```

should work.

Reply:
> not sure how it would be implemented with posix_spawn

This is a good example of why it's necessary to decide up-front whether we are allowing ourselves to be restricted by the limitations of `posix_spawn` or not!

Reply:
On modern glibc there is `posix_spawn_file_actions_addchdir_np()`, but (as with many other nice things) it's a non-portable GNU extension.